### DISCUSSION TITLE
增加数学符号输入支持

### DISCUSSION THREAD

**Discussion by yjfvictor** [rime/home#720] [Ideas]
能否考虑增加数学符号输入支持？
这个东西在Linux上的fcitx输入法框架的unicode-latex输入方案已经支持了，希望有一个rime版本的。

这里有码表可用
https://github.com/hubutui/fcitx-table-unicode-latex

MIT许可证，无版权问题，可以直接拿来，转一下码表格式就能用。

**Comment by yjfvictor** [comment]
码表已转成rime格式，可以加入项目了吧
[unicode_latex.zip](https://github.com/rime/plum/files/4872628/unicode_latex.zip)

**Comment by lotem** [comment]
不錯不錯。
請建立一個公開的GitHub代碼倉庫，像[這樣](https://github.com/lotem/rime-zhengma)把YAML文件放在裏面，就可以通過 rime-install 命令安裝了。

**Comment by shenlebantongying** [comment]
他已经很久没有回复了，我直接建了一个仓库并大幅度修改了一下：

https://github.com/shenlebantongying/rime_latex

我有一个问题。码表里面的 [columns](https://github.com/LEOYoon-Tsaw/Rime_collections/blob/master/Rime_description.md#%E7%A4%BA%E4%BE%8B-14) 中 stem 和 code 有什么区别？


我想为一些符号添加上别称，是不是只要创建两条有同样 text 但是不同 code 的就可以？

**Comment by lotem** [comment]
`code`即用於輸入該詞條的編碼，`stem`是字形輸入法以該條目的單字構造詞組時所採取的編碼（字根）。

**Comment by lotem** [comment]
於是說，現在樓主提出的這個項目已經完成了？可以關閉問題了嘛？

**Comment by shenlebantongying** [comment]
可以用这个输入 latex，符号是以 `\` 开头

https://github.com/shenlebantongying/rime_latex

另外 README 里面有配合其它输入法的说明。

![psc](https://user-images.githubusercontent.com/20123683/95442479-6e134d80-0929-11eb-9c54-1f2d4ad7e5ea.jpg)
![psc](https://user-images.githubusercontent.com/20123683/95443008-0b6e8180-092a-11eb-930b-a746cd015869.jpg)

**Comment by iningz** [comment]
@shenlebantongying 請問能否詳細解釋下如何將其作爲其他輸入法的輔助？我直接用東風破安裝並在小鶴雙拼加入patch後沒有效果，而且easy-en也失效了。

**Comment by shenlebantongying** [comment]
@Yining765 由於我只在寫英語的時候需要用 Latex-math, 所以一直都是單獨用。。。

研究過 patch 的機制，但是後來沒時間就沒弄完，現在暫時只能單獨用。。。

如果你想現在用，可以直接把我倉庫裏面的 latex.dict.yaml 文件後面的符號表部分直接複製到 easy_en.dict.yaml 後面。

====

更完美的方法應該有，不清楚我什麼時候有時間可以弄:

1 簡土方法：讓 rime 吧數學符號當作“標點符號”來處理

https://github.com/rime/rime-prelude/blob/master/symbols.yaml#L80
![image](https://user-images.githubusercontent.com/20123683/108221958-52555a80-7106-11eb-9351-2e5a459cc276.png)

2 理論上更好的辦法是添加一個 speller，segmentor,以 `\` 開頭的部分作爲一個單元來處理，同時可以自動上屏

可以參考這個人做的 .dict 和 .schema
https://github.com/jakwings/rime-xyz/blob/cffb2e0a90059986a356675c0e01f1af73f7cbca/symbols.schema.yaml
https://github.com/jakwings/rime-xyz/blob/cffb2e0a90059986a356675c0e01f1af73f7cbca/symbols.dict.yaml

3 opencc https://github.com/sgalal/rime-symbolic

**Reply by YaoLiMuMu** [reply]
方法2是完美的, 可惜大家都没空整了

**Reply by shenlebantongying** [reply]
https://github.com/shenlebantongying/rime_latex

仓库 README 里有写一个可用的方法
